dev/src/compiler.cpp

```diff
@@ -288,80 +293,167 @@ string HTMLEscape(string_view in) {
             for (auto [i, a] : enumerate(nf->retvals)) {
-                s += "<font color=\"#666666\">";
-                s += HTMLEscape(TypeName(a.type, a.fixed_len));
-                s += "</font>";
+                s += docTags[Tags::RetTypeWrap][0];
+                s += docTags[Tags::Font][0];
+                s += escape(TypeName(a.type, a.fixed_len));
+                s += docTags[Tags::Font][1];
+                s += docTags[Tags::RetTypeWrap][1];
                 if (i < nf->retvals.size() - 1) s += ", ";
             }
+            s += docTags[Tags::Returns][1];
         }
-        s += cat("</tt></td><td class=\"a\">", HTMLEscape(nf->help), "</td></tr>\n");
+        s += cat(docTags[Tags::Help][0], escape(nf->help), docTags[Tags::Help][1], "\n");
+        s += docTags[Tags::Row][1];
+        is_first_row = false;
     }
-    s += "</table>\n</td></tr></table></center></body>\n</html>\n";
+    s += docTags[Tags::Table][1];
+    s += docTags[Tags::Doc][1];
+    return s;
+}
+
+void DumpBuiltinDoc(NativeRegistry &nfr, bool group_subsystem) {
+    string htmlTags[16][2] = {
+    /* Doc          */  {"<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2 Final//EN\">\n"                                                        
+    /*              */   "<html>\n<head>\n<title>lobster builtin function reference</title>\n"
+    /*              */   "<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" />\n"
+    /*              */   "<style type=\"text/css\">"
+    /*              */   "table.a, tr.a, td.a {font-size: 10pt;border: 1pt solid #DDDDDD;"
+    /*              */   " border-Collapse: collapse; max-width: 88em}</style>\n"
+    /*              */   "</head>\n<body><center><table border=0><tr><td>\n<p>"
+    /*              */   "lobster builtin functions:"
+    /*              */   "(file auto generated by compiler, do not modify)</p></td></tr>", "\n</table></center></body>\n</html>\n"},
+    /* Table        */  {"<tr><td><table class=\"a\" border=1 cellspacing=0 cellpadding=4>", "</table></td></tr>\n"},
+    /* Row          */  {"<tr class=\"a\" valign=top>", "</tr>\n"},
+    /* FirstRow     */  {"<tr class=\"a\" valign=top>", "</tr>\n"},
+    /* Td           */  {"<td>", "</td>"},
+    /* Subsystem    */  {"<td><h3>", "</h3></td>"},
+    /* Name         */  {"<td class=\"a\"><tt><b>", "</b>"},
+    /* Params       */  {"(", ")"},
+    /* Help         */  {"</tt></td><td class=\"a\">", "</td>\n"},
+    /* Font         */  {"<font color=\"#666666\">", "</font>"},
+    /* Returns      */  {" -> ", ""},
+    /* ParamType    */  {": ", ""},
+    /* ParamName    */  {"", ""},
+    /* ParamDefault */  {" = ", ""},
+    /* Param        */  {"", ""},
+    /* RetTypeWrap  */  {"", ""}};
+    string s = GetBuiltinDoc(nfr, group_subsystem, htmlTags, HTMLEscape);
     WriteFile("builtin_functions_reference.html", false, s, false);
 }
 
+void DumpBuiltinDocJson(NativeRegistry &nfr) {
+    string jsonTags[16][2] = { 
```

Comment:
this codebase doesn't use `camelCase` anywhere, for vars it be `snake_case`, please fix all new identifiers in this PR.